plugins/vina.py

```diff
@@ -0,0 +1,1086 @@
+"""
+    Vina Plugin for PyMOL
+    
+    Pedro Sousa Lacerda <pslacerda@gmail.com>
+    LaBiMM: Laboratório de Bioinformática e Modelagem Molecular
+    
+    Be sure to adapt the DEFAULT_PREFS for your needs. It is defaulted to
+    Debian/Ubuntu Linux.
+"""
+
+import os
+from os.path import (
+    expanduser,
+    dirname,
+    splitext,
+    basename,
+)
+from glob import glob
+import itertools
+from operator import itemgetter
+import subprocess
+from tempfile import mkstemp
+import shutil
+import shlex
+import textwrap
+import subprocess
+import json
+from contextlib import contextmanager
+from urllib.parse import urlencode
+from urllib.request import Request, urlopen
+
+import pymol
+import pymol.gui
+from pymol import cmd
+from pymol.cgo import CYLINDER, SPHERE, COLOR, cyl_text
+from pymol.vfont import plain
+import numpy as np
+
+QWidget = pymol.Qt.QtWidgets.QWidget
+QFileDialog = pymol.Qt.QtWidgets.QFileDialog
+QFormLayout = pymol.Qt.QtWidgets.QFormLayout
+QPushButton = pymol.Qt.QtWidgets.QPushButton
+QSpinBox = pymol.Qt.QtWidgets.QSpinBox
+QDoubleSpinBox = pymol.Qt.QtWidgets.QDoubleSpinBox
+QDockWidget = pymol.Qt.QtWidgets.QDockWidget
+QLineEdit = pymol.Qt.QtWidgets.QLineEdit
+QCheckBox = pymol.Qt.QtWidgets.QCheckBox
+QApplication = pymol.Qt.QtWidgets.QApplication
+QMessageBox = pymol.Qt.QtWidgets.QMessageBox
+QVBoxLayout = pymol.Qt.QtWidgets.QVBoxLayout
+QTextEdit = pymol.Qt.QtWidgets.QTextEdit
+QDialog = pymol.Qt.QtWidgets.QDialog
+QDialogButtonBox = pymol.Qt.QtWidgets.QDialogButtonBox
+QDesktopWidget = pymol.Qt.QtWidgets.QDesktopWidget
+QProgressBar = pymol.Qt.QtWidgets.QProgressBar
+QAction = pymol.Qt.QtWidgets.QAction
+QComboBox = pymol.Qt.QtWidgets.QComboBox
+
+LeftDockWidgetArea = pymol.Qt.QtCore.Qt.LeftDockWidgetArea
+QRegExp = pymol.Qt.QtCore.QRegExp
+QtCore = pymol.Qt.QtCore
+QThread = pymol.Qt.QtCore.QThread
+pyqtSignal = pymol.Qt.QtCore.pyqtSignal
```

Comment:
This will work with PyQt and PySide:
```python
pyqtSignal = pymol.Qt.QtCore.Signal
```
See https://github.com/schrodinger/pymol-open-source/blob/445bf1a42c222cf1ac3a95c51699d81380698c29/modules/pymol/Qt/__init__.py#L85